src/dmd/escape.d

```diff
@@ -1110,6 +1109,26 @@ bool checkNewEscape(Scope* sc, Expression e, bool gag)
         // Only look for errors if in module listed on command line
         if (p == sc.func)
         {
+
+            // https://issues.dlang.org/show_bug.cgi?id=21525
+            if (sc.func.fes && v.storage_class & STC.parameter)
+            {
+                /* If we are inside a foreach body, the ref parameter
+                 * to the foreach generated delegate should still be
+                 * able to escape the foreach scope
+                 */
+                bool isForeachParam;
+                foreach(ref param; *(sc.func.parameters))
+                {
+                    if (param == v)
+                    {
+                        isForeachParam = true;
```

Comment:
Instead of the flag, replace this line with:
```
continue Loop;
```
and add `Loop:` to the enclosing `foreach`. Eliminating such flags is the point of the `continue _Identifier_` syntax.